### PR TITLE
Add integration with Grafana Agent

### PR DESCRIPTION
**Pull request by samuelallan72**
Add integration with Grafana Agent

- pull in the cos_agent interface library
- configure the relation with path and port for openstack-exporter metrics

## Testing

Some helpful commands for testing:

```
charmcraft -v pack
juju deploy ./openstack-exporter_ubuntu-22.04-amd64.charm --resource openstack-exporter=golang-openstack-exporter_1.6.0_amd64.snap
juju deploy grafana-agent
juju relate openstack-exporter grafana-agent
```

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -46,6 +47,13 @@ def __init__(self, *args: tuple[Any]) -> None:
         """Initialize the charm."""
         super().__init__(*args)
 
+        self._grafana_agent = COSAgentProvider(
```
Comment (chanchiwai-ray) [inline src/charm.py:50]: It's worth thinking should we start the snap service without this relation. We generally think the exporter service should be started only when it's required (i.e. there is consumers for the metric). Can you add some logic in the `configure` method to properly handle that?

**Review by chanchiwai-ray — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (chanchiwai-ray) [inline CONTRIBUTING.md:1]: Thanks for the doc updates, can you move this to a separate PR? I need these to create workflow for automatic updating the libs

File: update_charm_libs.sh
Comment (chanchiwai-ray) [inline update_charm_libs.sh:1]: Also move this to a new PR

**Review by chanchiwai-ray — CHANGES_REQUESTED** [review]

**Review by chanchiwai-ray — DISMISSED** [review]
LGTM, thanks

File: CONTRIBUTING.md
Comment (chanchiwai-ray) [reply CONTRIBUTING.md:1]: Thanks

**Review by jneo8 — APPROVED** [review]

**Review by chanchiwai-ray — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

**Pull request merged**